linker/shared/src/main/scala/org/scalajs/linker/checker/ClassDefChecker.scala

```diff
@@ -93,6 +94,9 @@ private final class ClassDefChecker(classDef: ClassDef, reporter: ErrorReporter)
     if (classDef.kind == ClassKind.ModuleClass &&
         methods(MemberNamespace.Constructor.ordinal).size != 1)
       reportError("Module class must have exactly 1 constructor")
+
+    if (classDef.kind.isJSClass && classDef.memberDefs.count(_.isInstanceOf[JSConstructorDef]) != 1)
```

Comment:
I had not thought about that. Unfortunately it doesn't work because `JS*` stuff does not make it into `methods()`. `methods(i)` is a `Set[MethodName]`, so it only contains `MethodDef`s. I could still iterate on `memberDefs` and check whether their namespace is `Constructor`, but it doesn't seem any better than directly checking for `JSConstructorDef`, does it?